tdms/include/arrays/incident_field.h

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file incident_field.h
+ * @brief Declaration for the IncidentField object.
+ */
+#pragma once
+
+#include <string>
+
+#include "arrays/tensor3d.h"
+#include "matrix.h"
+
+class IncidentField {
```

Comment:
Could do with a docstring if you have it in your mind.
Although I'm fine if not: that that's a bit of a scope-drift for this PR.